File: Estrutura de Decisao/Exercicio 1/CalculaDoisNumeros.php
<?php
/*
Faça um Programa que leia 2 números e em seguida pergunte ao usuário qual operação ele deseja realizar.
O resultado da operação deve ser acompanhado de uma frase que diga se o número é:
par ou ímpar;
positivo ou negativo;
inteiro ou decimal.
*/
$a = readline("Insira o primeiro número: ");
$b = readline("Insira o segundo número: ");

echo PHP_EOL . '[1] Soma || [2] Subtrai || [3] Divide || [4] - Multiplica' . PHP_EOL;

$decisao = intval(readline("O que você deseja fazer?  "));

switch ($decisao) {
    case 1:
        $resultado = $a + $b;
        descricao($resultado);
        break;
    case 2:
        $resultado = $a - $b;
        descricao($resultado);
        break;
    case 3:
        $resultado = $a / $b;
        descricao($resultado);
        break;
    case 4:
        $resultado = $a * $b;
        descricao($resultado);
        break;
}

function descricao($numero)
{
    $frase = "O número " . $numero . " é:";
    if ($numero % 2 != 0) {
        $frase = $frase . PHP_EOL . "Ímpar;";
    } else {
        $frase = $frase . PHP_EOL . "Par;";
    }

    if ($numero < 0) {
        $frase = $frase . PHP_EOL . 'Negativo;';
    } else {
        $frase = $frase . PHP_EOL . 'Positivo;';
    }

    if (!is_float($numero)) {
        $frase = $frase . PHP_EOL . 'Inteiro;';
    } else {
        $frase = $frase . PHP_EOL . 'Decimal;';
    }

    echo $frase;
}

File: Estrutura de Decisao/Exercicio 1/CalculaDoisNumeros.py
'''
Faça um Programa que leia 2 números e em seguida pergunte ao usuário qual operação ele deseja realizar.
O resultado da operação deve ser acompanhado de uma frase que diga se o número é:
par ou ímpar;
positivo ou negativo;
inteiro ou decimal.
'''


def descreve(numero):
    numeroParaTexto = str(numero)
    frase = "O número " + numeroParaTexto + " é:"
    if (numero % 2 != 0):
        frase = frase + "\nÍmpar;"
    else:
        frase = frase + "\nPar;"

    if numero < 0:
        frase = frase + '\nNegativo;'
    else:
        frase = frase + '\nPositivo;'

    if numero // 1 == numero:
        frase = frase + '\nInteiro;'
    else:
        frase = frase + '\nDecimal;'

    print(frase)


a = float(input("Insira o primeiro número:\n"))
b = float(input("Insira o segundo número:\n"))

decisao = int(input(
    "O que você deseja fazer?\n[1] Soma\n[2] Subtrai\n[3] Divide\n[4] Multiplicação\n\n"))

if decisao == 1:
    resultado = a + b
    descreve(resultado)
elif decisao == 2:
    resultado = a - b
    descreve(resultado)
elif decisao == 3:
    resultado = a / b
    descreve(resultado)
elif decisao == 4:
    resultado = a*b
    descreve(resultado)
